src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -269,6 +269,7 @@ private static void updateAuthorList(RepoConfiguration config) throws NoAuthorsW
 
             config.setAuthorList(authorList);
         }
+        config.removeIgnoredAuthors();
```

Comment:
This doesn't really go along well with this method because the purpose of this method is to 
```
Find and update {@code config} with all the author identities if author list is empty.
```
and not really to serve as a post processing mechanism as things done in this method are meant to be conditional.

In fact, we should probably make the ignore list a "block entry" or "hiding" mechanism in the first place.
E.g. prevent adding/setting authors through RepoConfiguration if the author is found in the ignore list

Reply:
> In fact, we should probably make the ignore list a "block entry" or "hiding" mechanism in the first place.
> E.g. prevent adding/setting authors through RepoConfiguration if the author is found in the ignore list

This is hard to do because at the time we add the authors found in `author-config.csv`, we have not processed StandaloneConfig JSON yet, and the ignore list might be specified in the JSON.

Reply:
> This doesn't really go along well with this method because the purpose of this method is to
> 
> ```
> Find and update {@code config} with all the author identities if author list is empty.
> ```
> 
> and not really to serve as a post processing mechanism as things done in this method are meant to be conditional.
> 

Makes a lot of sense, thanks for pointing that out. I will move it to a higher level.

Reply:
I personal agrees to put the method here and update the doc.